LandSandBoat/server
# [lua] Add generic logic for Mog Kupons, and data for first 5 kupons

**_I affirm:_**
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

## What does this PR do?

Adds the driver code for handling kupon items and KIs, and adds the data for the Dyna Beau + Xarc kupons. Others won't do anything until the data is entered.

## How to test

- Go to dealer moogle in windy: !pos -65.377 -4.188 144.133 240
- Speak to them
- Try to trade garbage to them - nothing happens
- Give yourself Dynamis - Beaucedine kupon: !additem 2745
- Speak to moogle again - nothing
- Trade them the Dynamis - Beaucedine kupon
- Select your chosen gift
- Get the item
- You should have no kupon item or key item, only the prize
- Give yourself Dynamis - Xarc kupon: !additem 2746
- Trade the kupon
- Bail out of the menu
- Get given the dynamis xarc kupon KI
- Talk to the moogle again, be offered the prize for that kupon
- bail out again - nothing happens, you still have the KI
- Fill your inventory
- Talk to moogle, try to get prize, it should fail, you should still have the KI
- Clear your inv
- talk to moogle, get prize, KI should be gone
- You should have 2 bits of prize gear now, no kupon items, no kupon KIs
